sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2850,15 +3058,32 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
    * }}}
    */
   override def visitReplaceTable(ctx: ReplaceTableContext): LogicalPlan = withOrigin(ctx) {
-    val (table, _, ifNotExists, external) = visitReplaceTableHeader(ctx.replaceTableHeader)
+    val (table, temp, ifNotExists, external) = visitReplaceTableHeader(ctx.replaceTableHeader)
+    if (temp) {
+      operationNotAllowed(
+        "CREATE OR REPLACE TEMPORARY TABLE ..., use CREATE TEMPORARY VIEW instead",
+        ctx)
+    }
+
     if (external) {
```

Comment:
shall we reconsider it? How can we re-create an external table?

Reply:
nvm, let's forbid it for now. `REPLACE EXTERNAL TABLE` looks ambiguous. It's not clear if the existing table is EXTERNAL, or we want to create an EXTERNAL TABLE to replace the existing one.